tests/functional/commands/test_db_load.py

```diff
@@ -0,0 +1,462 @@
+    assert collections == {"experiments", "algo", "trials", "benchmarks"}
+
+    assert len(loaded_db.read("benchmarks")) == 3
+    assert len(loaded_db.read("experiments")) == 3
+    assert len(loaded_db.read("trials")) == 24
+    # TODO: We should expect 6 algorithms, but only 3 are returned
+    # It seems config `three_experiments_family_same_name` contains 3 supplementary algorithms
+    # that are not related to experiments registered in the database. So, when dumping from this config
+    # then loading from dumped data, only algorithms related to available experiments are loaded,
+    # and there are only 3 such algorithms (1 per experiment)
+    assert len(loaded_db.read("algo")) == 3
+
+
+def test_load_again_without_resolve(empty_database, capsys):
+    """Test load all database twice without resolve in second call"""
+    storage = setup_storage()
+    loaded_db = storage._db
+    assert len(loaded_db.read("benchmarks")) == 0
+    assert len(loaded_db.read("experiments")) == 0
+    assert len(loaded_db.read("trials")) == 0
+    assert len(loaded_db.read("algo")) == 0
+
+    execute(f"db load {LOAD_DATA_WITH_BENCHMARKS}")
+    benchmarks = loaded_db.read("benchmarks")
+    experiments = loaded_db.read("experiments")
+    trials = loaded_db.read("trials")
+    algos = loaded_db.read("algo")
+    assert len(benchmarks) == 3
+    assert len(experiments) == 3
+    assert len(trials) == 24
+    assert len(algos) == 3
+
+    # Again
+    execute(f"db load {LOAD_DATA_WITH_BENCHMARKS}", assert_code=1)
+    captured = capsys.readouterr()
+    assert (
+        captured.err.strip()
+        == "Error: Conflict detected without strategy to resolve (None) for benchmark branin_baselines_webapi"
+    )
+    # Destination should have not changed
+    new_benchmarks = loaded_db.read("benchmarks")
+    new_experiments = loaded_db.read("experiments")
+    new_trials = loaded_db.read("trials")
+    new_algos = loaded_db.read("algo")
+    assert new_benchmarks == benchmarks
+    assert new_experiments == experiments
+    assert new_trials == trials
+    assert new_algos == algos
+
+
+def test_load_ignore(empty_database):
+    """Test load all database with --resolve ignore"""
+    storage = setup_storage()
+    loaded_db = storage._db
+    assert len(loaded_db.read("benchmarks")) == 0
+    assert len(loaded_db.read("experiments")) == 0
+    assert len(loaded_db.read("trials")) == 0
+    assert len(loaded_db.read("algo")) == 0
+
+    execute(f"db load {LOAD_DATA_WITH_BENCHMARKS}")
+    benchmarks = loaded_db.read("benchmarks")
+    experiments = loaded_db.read("experiments")
+    trials = loaded_db.read("trials")
+    algos = loaded_db.read("algo")
+    assert len(benchmarks) == 3
+    assert len(experiments) == 3
+    assert len(trials) == 24
+    assert len(algos) == 3
+
+    execute(f"db load {LOAD_DATA_WITH_BENCHMARKS} -r ignore")
+    # Duplicated data should be ignored, so we must expect same number of data and same IDs.
```

Comment:
As these tests stand, it is not possible to distinguish the effect of no resolution, ignore or overwrite. At least one document of each collection should be modified so that we can verify that the modified version was not used in case of ignore, or was used in case of overwrite.
You could modify harmless fields like `experiment.algorithm`, `trial.status` and `algo.hearthbeat` for these tests. These would have no impact of the IDs.

Reply:
Tested!